src/state_point.cpp

```diff
@@ -544,37 +545,64 @@ write_source_point(const char* filename)
   }
 
   // Get pointer to source bank and write to file
-  write_source_bank(file_id);
+  write_source_bank(file_id, surf_src_bank);
 
   if (mpi::master || parallel) file_close(file_id);
 }
 
 void
-write_source_bank(hid_t group_id)
+write_source_bank(hid_t group_id, bool surf_src_bank)
 {
   hid_t banktype = h5banktype();
 
+  // Set total and individual process dataspace sizes for source bank
+  int64_t dims_size = settings::n_particles;
+  int64_t count_size = simulation::work_per_rank;
+
+  // Set maximum bank size
+  simulation::max_bank_size = simulation::work_per_rank;
+
+  // Set vectors for source bank and starting bank index of each process
+  std::vector<int64_t> bank_index = simulation::work_index;
+  std::vector<Particle::Bank> src_bank = simulation::source_bank;
```

Comment:
Because of scope and conditionals, we need to a little careful where things are defined.  I think the right psuedo-code is:

```
std:vector<Particle::Bank> *src_bank_ptr = &simulation::source_bank;
std:vector<Particle::Bank> surf_src_bank_vector;

if (surf_src_bank) {
    surf_src_bank_vector.resize(max_bank_size)
    surf_src_bank_vector.assing(....)

    (etc)

    src_bank_ptr = &surf_src_bank_vector;

}
```